From 82cf6f8fca841a5a6cebfcb37a2d6acc305b08d4 Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Wed, 18 Feb 2015 16:36:00 +0530
Subject: [PATCH] FG/MAXIM: Updated the fg config acpi header.

In an acpi based systems, the fg config data will be provided thru BCFG
acpi table from bios. Driver should read the fg data from BCFG table
and initialize the fg hardware. This patch will update the fg config
header in the driver to the header used in bios.

Change-Id: I1a5ef6f01cd5a44191ad2b5e532ae7c2f559db9e
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1044
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
---
 drivers/power/max17042_battery.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/power/max17042_battery.c b/drivers/power/max17042_battery.c
index 421a9cb..18e18de 100644
--- a/drivers/power/max17042_battery.c
+++ b/drivers/power/max17042_battery.c
@@ -946,9 +946,10 @@ struct max170xx_acpi_fg_config {
 	char fg_name[ACPI_FG_NAME_LEN];
 	char battid[BATTID_LEN];
 	u16 size;
+	u8 fco; /* Fuel gauge config options */
 	u16 checksum;
 	struct max17042_config_data cdata;
-};
+} __packed;
 
 static struct max17042_config_data *
 max17042_get_acpi_cdata(struct device *dev)
-- 
1.9.1

